src/shop-list.html

```diff
@@ -98,11 +107,13 @@
     </header>
 
     <div class="grid" hidden$="[[failure]]">
+      <ul>
```

Comment:
maybe make this element `<ul class="grid" hidden$="[[failure]]">` and remove the outer div element.